docs/addons/external-dns.md

```diff
@@ -0,0 +1,157 @@
+# External DNS Add-on
+
+External DNS add-on is based on the [ExternalDNS](https://github.com/kubernetes-sigs/external-dns) open source project and allows integration of exposed Kubernetes services and Ingresses with DNS providers, in particular [Amazon Route 53](https://aws.amazon.com/route53/).
+
+The add-on provides functionality to configure IAM policies and Kubernetes service accounts for Route 53 integration support based on [AWS Tutorial for External DNS](https://github.com/kubernetes-sigs/external-dns/blob/master/docs/tutorials/aws.md).
+
+## Usage
+
+```typescript
+import * as ssp from '@shapirov/cdk-eks-blueprint';
+
+readonly externalDns = new ssp.addons.ExternalDnsAddon({
+    hostedZone: new ssp.addons.DelegatingHostedZoneProvider(
+        parentDomain,
+        subdomain, 
+        parentDnsAccountId,
```

Comment:
does this addon require a parentDnsAccountId? What about hosted zones with the parentDomain in the same account?